examples/cd_mol.ipynb

```diff
@@ -0,0 +1,1071 @@
+    "\n",
+    "We will first detect drift on the prediction distribution of the GIN model. Since the binary classification model returns continuous numerical univariate predictions, we use the [Kolmogorov-Smirnov drift detector](https://docs.seldon.io/projects/alibi-detect/en/latest/methods/ksdrift.html). First we define some utility functions:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 17,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from torch_geometric.data import Batch, Data\n",
+    "from typing import Dict, List\n",
+    "\n",
+    "\n",
+    "labels = ['No!', 'Yes!']\n",
+    "def make_predictions(dd, xs: Dict[str, List[Data]]) -> None:\n",
+    "    for split, x in xs.items():\n",
+    "        preds = dd.predict(x)\n",
+    "        dl = DataLoader(x, batch_size=32, shuffle=False)\n",
+    "        _ = evaluate(dl, split)\n",
+    "        print('Drift? {}'.format(labels[preds['data']['is_drift']]))\n",
+    "        if isinstance(preds[\"data\"][\"p_val\"], (list, np.ndarray)):\n",
+    "            print(f'p-value: {preds[\"data\"][\"p_val\"]}')\n",
+    "        else:\n",
+    "            print(f'p-value: {preds[\"data\"][\"p_val\"]:.3f}')\n",
+    "        print('')\n",
+    "        \n",
+    "\n",
+    "def sample(split: str, n: int, ) -> List[Data]:\n",
+    "    idx = np.random.choice(split_idx[split].numpy(), size=n, replace=False)\n",
+    "    return [dataset[i] for i in idx]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Because we pass lists with `torch_geometric.data.Data` objects to the detector, we need to preprocess the data using the `batch_fn` into `torch_geometric.data.Batch` objects which can be fed to the model. Then we detect drift on the model prediction distribution."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 18,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "WARNING:alibi_detect.cd.utils:Input shape could not be inferred. If alibi_detect.models.tensorflow.embedding.TransformerEmbedding is used as preprocessing step, a saved detector cannot be reinitialized.\n"
+     ]
+    }
+   ],
+   "source": [
+    "from alibi_detect.cd import KSDrift\n",
+    "from alibi_detect.utils.pytorch import predict_batch\n",
+    "from functools import partial\n",
+    "\n",
+    "\n",
+    "def batch_fn(data: List[Data]) -> Batch:\n",
+    "    return Batch().from_data_list(data)\n",
+    "\n",
+    "\n",
+    "preprocess_fn = partial(predict_batch, model=model, device=device, preprocess_fn=batch_fn, batch_size=32)\n",
+    "dd = KSDrift(x_ref, p_val=.05, preprocess_fn=preprocess_fn)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Since the dataset is heavily imbalanced, we will test the detectors on a sample which oversamples from the minority class (molecules which inhibit HIV virus replication):"
```

Comment:
This implies our detectors do not work on class-imbalanced data. I assume that's not the case?

Reply:
I think I misunderstood this. The correction of the class imbalance is introduced as a manifestation of drift? I think it might help to make this clearer.

Reply:
Yes that's correct, I will clarify this sentence.